src/clr-angular/data/datagrid/datagrid-pagination.ts

```diff
@@ -69,6 +71,7 @@ export class ClrDatagridPagination implements OnDestroy, OnInit {
      */
     private _pageSubscription: Subscription;
     ngOnDestroy() {
+        this.pageSize = 0;
```

Comment:
While this works, I personally would rather there be a method on the page service to reset so the intention here is more explicit.

Reply:
yes. thats a good idea. i can make that change. will just add a method to the page service and set pageSize to 0